src/components/AudioPlayer/AudioPlayer.tsx

```diff
@@ -0,0 +1,192 @@
+        return audioType !== "file" && doc?.src?.includes(audioType);
+      }
+
+      if (iframeTitle && audioType === "file") {
+        console.warn(
+          'NYPL Reservoir AudioPlayer: The `iframeTitle` prop will be ignored when the `audioType` prop is set to "file."'
+        );
+      }
+      // Only set the iframe title if it doesn't already have it in the iframe
+      if (iframeDoc && !iframeDoc.title) {
+        iframeDoc.title = iframeTitle ? iframeTitle : "Embedded audio player";
+      }
+
+      // A friendly message states that we don't support filePath prop
+      const isLocalAudioFile = filePath || audioType === "file";
+      if (isLocalAudioFile) {
+        console.warn(
+          "The current AudioPlayer does not support locally hosted audio file."
+        );
+      }
+
+      const isLocalWithEmbed = audioType === "file" && embedCode;
+      if (isLocalWithEmbed) {
+        // Question: should we show this now?
+        console.warn(
+          'NYPL Reservoir AudioPlayer: The `embedCode` prop will be ignored when the audioType prop is set to "file".'
+        );
+      }
+      const isLocalWithoutPath = audioType === "file" && !filePath;
+      if (isLocalWithoutPath) {
+        // Question: Should we remove this warning since we don't support filepath?
+        console.warn(
+          'NYPL Reservoir AudioPlayer: The `filePath` prop is required when the `audioType` prop is set to "file."'
+        );
+      }
+
+      const isThirdPartyService = !!thirdPartyServices.find(
+        (service) => service === audioType
+      );
+      const isThirdPartyAndLocal = isThirdPartyService && filePath;
+      if (isThirdPartyAndLocal) {
+        console.warn(
+          "NYPL Reservoir AudioPlayer: The `filePath` prop will be ignored when using a 3rd party streaming service."
+        );
+      }
+
+      const isThirdPartyWithoutCode = isThirdPartyService && !embedCode;
+      if (isThirdPartyWithoutCode) {
+        console.warn(
+          "NYPL Reservoir AudioPlayer: The `embedCode` prop is required when using a 3rd party streaming service."
+        );
+        isInvalidEmbed = true;
+      }
+
+      const styles = useMultiStyleConfig("AudioPlayer", {});
+
+      /**
+       * Builder function to build the finalized render element, we prioritize localfile over 3rd party code
+       * but for the current version, we only use 3rd party code.
+       */
+      function buildFinalElement(): JSX.Element {
+        return iframeDoc ? (
+          <Box
+            dangerouslySetInnerHTML={{
+              __html: iframeDoc.outerHTML,
+            }}
+          />
+        ) : null;
+      }
+
+      const embedElement = buildFinalElement();
+
+      return (
+        <Box
+          className={className}
+          data-testid="audio-player-component"
+          id={id}
+          ref={ref}
+          __css={styles}
+          {...rest}
+        >
+          <ComponentWrapper
```

Comment:
I'm wondering if you need the wrapper `Box` at all. Maybe you can pass all the props to `ComponentWrapper` and have that be the top level element. Does that work? If not, that's okay, just curious.

Reply:
The same structure is used in `VideoPlayer`.  If we find that the parent `Box` element can be removed, then the same change should be made to `VideoPlayer`.

Reply:
Yes, but that shouldn't be done in this PR if we decide to make that update happen. The other difference is that in this component, `ComponentWrapper` is always rendered, but in `VideoPlayer`, the `ComponentWrapper` is conditionally rendered. That's why I'm thinking that the `Box` can be removed and just always render the `ComponentWrapper`.

Here, if there's an error, the heading and description will render along with the error message. In the `VideoPlayer`, if there's an error, only the error message will render and not the heading or description. I think maybe we can rethink the `VideoPlayer` but, again, that should be a different PR.